File: GFG/Array Removals/Array Removals.java
class Solution {
    int removals(int[] arr, int n, int k) {
        // code here
        Arrays.sort(arr);
        int i=0;
        int j=0;
        int maxSize=0;
        while(j<n){
            if(arr[j]-arr[i]<=k)j++;
            else if(i<j)i++;
            maxSize=Math.max(maxSize,j-i);
        }
        return n-maxSize;
    }
}

File: GFG/Array Removals/Array Removals.cpp
class Solution{
    public:
    int removals(vector<int>& arr, int k){
        sort(arr.begin(), arr.end());
        
        int l = 0, r = 0, len = 0;
        
        while(r<arr.size()){
            if(arr[r]-arr[l]<=k){
                r++;
                len = max(len, r-l);
            }else l++;
        }
        
        return arr.size()-len;
    }
};
